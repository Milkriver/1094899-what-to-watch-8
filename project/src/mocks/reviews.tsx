const reviews = [
  {
    id: 1,
    movieId: 1,
    author: 'Kate Muir',
    date: '2016-12-24',
    rating: 8.9,
    text: 'Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director{&apos;}s funniest and most exquisitely designed films in years.',
  },
  {
    id: 2,
    movieId: 1,
    author: 'Bill Goodykoontz',
    date: '2015-11-18',
    rating: 8.0,
    text: 'Anderson&apos;s films are too precious for some, but for those of us willing to lose ourselves in them, they&apos;re a delight. &quot;The Grand Budapest Hotel&quot; is no different, except that he has added a hint of gravitas to the mix, improving the recipe.',
  },
  {
    id: 3,
    movieId: 1,
    author: 'Amanda Greever',
    date: '2015-11-18',
    rating: 8.0,
    text: 'I didn&apos;t find it amusing, and while I can appreciate the creativity, it&apos;s an hour and 40 minutes I wish I could take back.',
  },
  {
    id: 4,
    movieId: 1,
    author: 'Matthew Lickona',
    date: '2016-12-20',
    rating: 7.2,
    text: 'The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.',
  },
  {
    id: 5,
    movieId: 1,
    author: 'Paula Fleri-Soler',
    date: '2016-12-20',
    rating: 7.6,
    text: 'It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.',
  },
  {
    id: 6,
    movieId: 1,
    author: 'Paula Fleri-Soler',
    date: '2016-12-20',
    rating: 7.0,
    text: 'It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.',
  },
];

export default reviews;
