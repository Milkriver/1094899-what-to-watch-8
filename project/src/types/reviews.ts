export interface IReview {
    id: number,
    movieId: number,
    author: string,
    date: string,
    rating: number,
    text: string,
  }
