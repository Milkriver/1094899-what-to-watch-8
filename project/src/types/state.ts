import { AuthorizationStatus } from '../const';
import { IMovie } from './common';
import { IReviewResponse } from './reviews';

export interface IState {
  currentGenre: string
  movies: IMovie[]
  isDataLoaded: boolean
  authorizationStatus: AuthorizationStatus,
  movie: IMovie,
  sameMovies: IMovie[],
  reviews: IReviewResponse[],
  promoMovie: IMovie,
}
