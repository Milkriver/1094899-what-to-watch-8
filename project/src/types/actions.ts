import { AxiosInstance } from 'axios';
import { Action } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { IState } from './state';

export enum ActionType {
  ChangeGenre = 'movieList/ChangeGenre',
  LoadMovies = 'movieList/LoadMovies',
  LoadSingleMovie = 'movieCard/LoadSingleMovie',
  LoadActiveMovie = 'movieCard/LoadActiveMovie',
  LoadSameGenreMovies = 'movieCard/LoadSameGenreMovies',
  LoadPromoMovie = 'movieCard/LoadPromoMovie',
  LoadFavouriteMovies = 'userList/LoadFavouriteMovies',
  AddFavoriteMovie = 'userList/AddFavoriteMovie',
  RemoveFavoriteMovie = 'userList/RemoveFavoriteMovie',
  LoadReviews = 'movieCard/LoadReviews',
  AddReview = 'movieCard/AddReview',
  RequireAuthorization = 'user/requireAuthorization',
  RequireLogout = 'user/requireLogout',
  LoadUserData = 'user/loadUserData',
}

export type ThunkActionResult<R = Promise<void>> = ThunkAction<R, IState, AxiosInstance, Action>;

export type ThunkAppDispatch = ThunkDispatch<IState, AxiosInstance, Action>;
