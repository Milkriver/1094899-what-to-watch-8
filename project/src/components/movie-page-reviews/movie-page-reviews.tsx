import { IReview } from '../../types/reviews';

interface IProps {
  reviews: IReview[],
}

export function MoviePageReviews({ reviews }: IProps): JSX.Element {

  return (
    <div className="film-card__reviews film-card__row">
      <div className="film-card__reviews-col">
        {
          reviews.map((review) => (
            <div className="review" key={review.id}>
              <blockquote className="review__quote">
                <p className="review__text">{review.text}</p>
                <footer className="review__details">
                  <cite className="review__author">{review.author}</cite>
                  <time className="review__date" dateTime="2016-12-24">{review.date}</time>
                </footer>
              </blockquote>
              <div className="review__rating">{review.rating}</div>
            </div>),
          )
        }
      </div>
    </div >
  );
}
