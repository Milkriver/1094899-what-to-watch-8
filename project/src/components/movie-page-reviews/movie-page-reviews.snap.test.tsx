import { render } from '@testing-library/react';
import { MoviePageReviews } from './movie-page-reviews';
import { IReviewResponse } from '../../types/reviews';


const mockReviews: IReviewResponse[] = [{ id: 1, user: { id: 1, name: 'reviewer' }, rating: 5, comment: 'cool', date: '2021-01-01T23:59:59' }];

describe('Component: MoviePageReviews', () => {
  test('should render correctly', () => {
    const page = <MoviePageReviews reviews={mockReviews}/>;
    const { container } = render(page);
    expect(container).toMatchSnapshot();
  });
});
