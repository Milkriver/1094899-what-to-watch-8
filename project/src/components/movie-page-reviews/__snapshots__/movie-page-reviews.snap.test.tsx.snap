// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MoviePageReviews should render correctly 1`] = `
<div>
  <div
    class="film-card__reviews film-card__row"
  >
    <div
      class="film-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            cool
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              reviewer
            </cite>
            <time
              class="review__date"
              datetime="2016-12-24"
            >
              January 1, 2021
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          Good
        </div>
      </div>
    </div>
  </div>
</div>
`;
